spyder/workers/updates.py

```diff
@@ -225,16 +227,38 @@ def _progress_reporter(self, block_number, read_size, total_size):
     def _download_installer(self):
         """Donwload latest Spyder standalone installer executable."""
         tmpdir = tempfile.gettempdir()
-        is_full_installer = (is_module_installed('numpy') or
-                             is_module_installed('pandas'))
-        if os.name == 'nt':
-            name = 'Spyder_64bit_{}.exe'.format('full' if is_full_installer
-                                                else 'lite')
+
+        # If running on macOS under Rosetta, platform.machine will not be
+        # correct so check for ARM64 in version instead. All other
+        # platforms will be x86_64.
+        mach = "arm64" if "ARM64" in platform.version() else "x86_64"
+
+        if parse(self.latest_release_version) > parse("5"):
+            # conda-based installer name
+            if os.name == 'nt':
+                plat, ext = 'Windows', 'exe'
+            elif platform.system == 'Darwin':
+                plat, ext = 'macOS', 'pkg'
+            else:
+                plat, ext = 'Linux', 'sh'
+            name = f'Spyder-{plat}-{mach}.{ext}'
         else:
-            name = 'Spyder{}.dmg'.format('' if is_full_installer else '-Lite')
+            # 5.x installer name
+            is_full_installer = (is_module_installed('numpy') or
+                                 is_module_installed('pandas'))
+            if os.name == 'nt':
+                name = 'Spyder_64bit_{}.exe'.format(
+                    'full' if is_full_installer else 'lite'
+                )
+            else:
+                name = 'Spyder{}{}.dmg'.format(
+                    '_{mach}' if mach == 'arm64' else '',
```

Comment:
```suggestion
                    f'-{mach}' if mach == 'arm64' else '',
```

I think an f-string is missing here to do the string replacement and we should `-` instead of `_` according to the new naming set in `installers/macOS/setup.py`

Reply:
F-string missing seems correct, but if we use `-` instead of `_` here we'll need to rename and re-upload the ARM64 installers too (as well as tweak that bit in the website and docs links). It may not be pretty, but at this point sticking with `_` for this bit for one more version or so before fixing things properly in 6.x anyway seems like the most pragmatic choice, vs. making extra work for ourselves now (aside from updating it in the installer generation instead, as I suggest there).